### PR TITLE
(DOCSP-19077): Better Disambiguate Between Android Synced/Local Schema Changes

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
## Pull Request Info
Also incorporated semi-related https://jira.mongodb.org/browse/DOCSP-18798

### Jira

- https://jira.mongodb.org/browse/DOCSP-19077

### Staged Changes (Requires MongoDB Corp SSO)

- [Android SDK - Modify an Object Schema](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-19077/sdk/android/examples/modify-an-object-schema/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/android/examples/modify-an-object-schema.txt [hunk 1]
```diff
@@ -14,152 +14,181 @@ Modify an Object Schema - Android SDK
...
+      create a corresponding :ref:`migration function
+      <android-migration-function>` to move data from the original schema
+      to the updated schema.
 
-.. include:: /includes/note-modify-schema-properties-of-synced-realms.rst
+      .. note:: Versions Update on Realm Open
 
-.. note:: Versions Update on Realm Open
+         Assume that each schema change in this example occurs after
+         an application has used each version for some amount of time. New
```
Comment (MongoCaleb) [inline source/sdk/android/examples/modify-an-object-schema.txt:34]: This note is a bit confusing. I think this change clarifies it, but feel free to do as you will with it. 

> Assume that each schema change shown in the following example occurs after the application has been using the existing schema. The new schema version numbers apply only after you open the {+realm+} and explicitly specify the new version number. In other words, you can't specify version 3 without previously specifying and using versions 0, 1, and 2.

File: source/sdk/android/examples/modify-an-object-schema.txt [hunk 2]
```diff
@@ -14,152 +14,181 @@ Modify an Object Schema - Android SDK
    :depth: 2
```
Comment (MongoCaleb) [inline source/sdk/android/examples/modify-an-object-schema.txt:14]: I think this page overall would benefit from a numbered list, like we get with Includes: 
(1) Add, Delete, or Modify the Schema
(existing content)
(2) Apply the changes
(existing Migration function content)

File: source/sync/synced-schema-overview.txt [hunk 3]
```diff
@@ -33,9 +34,12 @@ updated need to work with the new schema changes.
 
 Additive Changes
 ----------------
+
 Additive changes, such as adding a class or adding a field to an existing class, 
 are applied automatically to synced {+realms+}; you can make these 
-additive schema changes with no additional configuration changes. 
+additive schema changes with no additional configuration changes. Increase
+the schema version in your client applications to use the new schema
```
Comment (MongoCaleb) [inline source/sync/synced-schema-overview.txt:41]: Can we make this a link to the page that describes this process?

Comment (nathan-contino-mongo) [reply source/sync/synced-schema-overview.txt:41]: Doing some extra digging with the SDK team on this process -- I'll likely handle that in a separate ticket. But definitely agreed!

**Pull request merged**